controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -594,17 +618,21 @@ func (r *KubeadmControlPlaneReconciler) reconcileExternalReference(ctx context.C
 		return err
 	}
 
+	if util.IsPaused(cluster, obj) {
+		return errors.Wrapf(&capierrors.RequeueAfterError{RequeueAfter: 30 * time.Second},
+			"Referenced %s %s is paused.",
+			obj.GetKind(),
+			obj.GetName())
+	}
+
 	patchHelper, err := patch.NewHelper(obj, r.Client)
 	if err != nil {
 		return err
 	}
 
-	obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), metav1.OwnerReference{
-		APIVersion: clusterv1.GroupVersion.String(),
-		Kind:       "Cluster",
-		Name:       cluster.Name,
-		UID:        cluster.UID,
-	}))
+	if err := controllerutil.SetControllerReference(kcp, obj, r.scheme); err != nil {
```

Comment:
This was a cluster owner before, and now it's KCP?

Reply:
That is correct. It feels more appropriate to me (at least now, compared to when I originally wrote this), that the infrastructure template should be "owned" by the KubeadmControlPlane vs the Cluster. Do you think that it's lifecycle should be tied to the Cluster instead?

Reply:
I was going to agree with you, but then IIRC, infra templates may be shared across MDs/MDs and across Clusters. I'm now leaning towards having you leave this as it was, unless I'm misremembering or misreading something. @vincepri keep me honest? 😄

Reply:
Yeah, if we follow the MachineSet controller, we should add a reference to the Cluster object and keep in mind there might be more than one reference to a Cluster.

https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/machineset_controller.go#L286-L291

Reply:
sgtm, I'll go ahead and update.